### DISCUSSION TITLE
Capturing Google Earth Pro/ Google mobile maps 3D buildings

### DISCUSSION THREAD

**Discussion by Mi-Pe** [eliemichel/MapsModelsImporter#136] [Ideas]
Hello,
I successfully captured google maps 3d scanned models from google chrome.
In big cities like Detroit google actually has two kinds of 3D models - the scanned ones and those probably made by hand for google mobile apps. They appear in mobile apps without textures and in google earth pro with textures, but they don't seem to appear in google chrome maps.
They have good topology and would be very useful for further editing.

This I guess is a screenshot from google maps mobile navigation:
https://forums.x-plane.org/uploads/monthly_2018_09/2016016860_Bildschirmfoto2018-09-29um22_38_56.jpg.8bb7edc73e4b214da63ecad604d2d333.jpg

This one is from google earth pro (don't mind some buildings flying):
![amełykańskie obiekty łatające](https://user-images.githubusercontent.com/73021367/105483714-89bd1c80-5caa-11eb-9ac4-6ba24c3148fb.jpg)

Do you think it would be possible to also capture these models and import them into Blender?

**Comment by eliemichel** [comment]
Interesting, do you have a RDC capture that contains some of these?

**Reply by Mi-Pe** [reply]
No, because I don't know how to do this :P
I only managed to capture and import the scanned model from google chrome but no success with google earth pro.
Actually I wanted to ask you if you think this would be possible to achieve. These models are much more accurate and could be used directly as a base for rendering. Very helpful for college student projects, for example.
I read and watched people using 3D Ripper XD for capturing 3d models from pretty much any app, but importing that to Blender was very troublesome.

**Reply by Mi-Pe** [reply]
Perhaps I could somehow read PID number of google earth pro, then inject the process into renderdoc and then export RDC capture. But I'm not enough tech-savvy to do it :P

**Reply by Mi-Pe** [reply]
Update: I found my Google Earth Pro PID number and injected the process into RenderDoc. Under "API" I can see "none". I'll see if I can do something with it. If you have any ideas yourself, please share.
I tried running Google Earth Pro both with directX and openGL and then injecting the process, but no luck. I've read somewhere that Google Earth Pro uses some old API that may be too old for RenderDoc. But it's all just guessing, I'm a noob in this kind of things.

**Reply by eliemichel** [reply]
Injection must occur before the app starts to use the graphics API, which is why when using Google Chrome I add a flag in the shortcut to make it show a popup. The point isn't so much to get the PID but to have the program pause between its start and its first use of the graphics API. Did you try starting Earth Pro from the Launch Application tab of RenderDoc?

**Reply by Mi-Pe** [reply]
Ok, I just tried it, typed Executable Path in RenderDoc and launched Google Earth Pro but right after launch it threw a prompt that there was a problem and it had to be shut.
Do I have to type some Command-line Arguments and/or Environment Variables in RenderDoc or use any of its numerous buttons to fix it? Do you know maybe?

**Reply by eliemichel** [reply]
I don't know, and there was the same issue when launching the browser. The whole point of injecting into the browser during the popup was designed to work this around but Earth Pro might not provide convenient command line arguments so it would require more advanced injection. I don't really have the time to look into this myself sorry.

**Reply by marty3d** [reply]
Hi @eliemichel 
First off, thank you for providing this amazing tool.

Is there any chance that you could look into this again? There are examples of google earth pro having geometry that is not available in the web browser versions. Example: Washington D.C. USA. The Mall area is missing from browser versions but is available in Pro.

**Reply by ScottSterIing** [reply]
@marty3d I was able to inject into Earth Pro and extract the geometry but from there I'm stuck, every time I attempt to import the .rdc into blender it doesn't actually import anything (.rdc's from web still import just fine)

(Import from Earth Pro) ![image](https://github.com/eliemichel/MapsModelsImporter/assets/134180939/446526e7-ad07-4fe1-991c-5594bbedddb9)

If you want the .rdc file from Pro I could send that to you, its over the limit atm

**Reply by Plutoaurus** [reply]
Hi.

How did you manage to inject into earth pro?  I get a opengl error so it can't be captured.

Thanks

**Reply by oreyre9000** [reply]
Me too. Pls wake up.

**Reply by ScottSterIing** [reply]
Oops, sorry I don't use github much, let me see if i can remember how I did it

@Plutoaurus @oreytv

**Reply by ScottSterIing** [reply]
@oreytv was able to inject and capture without any issue, whats holding it up for you?

**Reply by oreyre9000** [reply]
Thank you, it's ok. I don't need to solve it as i have instead resorted to using a blender plugin for my needs.

**Reply by ScottSterIing** [reply]
Yeah, I've stopped using the raw extracts as well, PlaceMaker (Sketchup) and Blosm (Blender) have been godsends recently

Sometimes it is nice to be able to have the geometry go into the distance but that is pretty niche scenarios.

**Comment by Mi-Pe** [comment]
Sure, don't worry. Thank you for letting me know :)
